numpy/linalg/tests/test_linalg.py

```diff
@@ -882,26 +882,24 @@ def test_axis(self):
 
         # Matrix norms.
         B = np.arange(1, 25, dtype=self.dt).reshape(2, 3, 4)
-
-        for order in [None, -2, 2, -1, 1, np.Inf, -np.Inf, 'fro', 'nuc']:
-            assert_almost_equal(norm(A, ord=order), norm(A, ord=order,
-                                                         axis=(0, 1)))
-
-            n = norm(B, ord=order, axis=(1, 2))
-            expected = [norm(B[k], ord=order) for k in range(B.shape[0])]
-            assert_almost_equal(n, expected)
-
-            n = norm(B, ord=order, axis=(2, 1))
-            expected = [norm(B[k].T, ord=order) for k in range(B.shape[0])]
-            assert_almost_equal(n, expected)
-
-            n = norm(B, ord=order, axis=(0, 2))
-            expected = [norm(B[:, k,:], ord=order) for k in range(B.shape[1])]
-            assert_almost_equal(n, expected)
-
-            n = norm(B, ord=order, axis=(0, 1))
-            expected = [norm(B[:,:, k], ord=order) for k in range(B.shape[2])]
-            assert_almost_equal(n, expected)
+        nd = B.ndim
+        for order in [None, -2, 2, -1, 1, np.Inf, -np.Inf, 'fro']:
+            for axis in itertools.combinations(range(-nd, nd), 2):
+                row_axis, col_axis = axis
+                if row_axis < 0:
+                    row_axis += nd
+                if col_axis < 0:
+                    col_axis += nd
+                if row_axis == col_axis:
+                    assert_raises(ValueError, norm, B, ord=order, axis=axis)
+                else:
+                    n = norm(B, ord=order, axis=axis)
+                    k_index = nd-(row_axis+col_axis)
```

Comment:
This seems to me a little too magic, I think it's worth a comment, indicating that it only works for `nd = 3`, and that the logic would have to be changed if that value is increased. Also, this could use some spaces around the `-` operator.